launcher/ui/themes/ThemeManager.h

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher
+ *  Copyright (C) 2022 Tayou <tayou@gmx.net>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <QString>
+#include <QIcon>
+#include <QDir>
+#include <QDirIterator>
+#include <QApplication>
+
+#include "ui/themes/ITheme.h"
+#include "ui/MainWindow.h"
+
+#define themeDebugLog qDebug() << "[Themes]"
+#define themeWarningLog qWarning() << "[Themes]"

```

Comment:
I thought this would be a neat way of doing it, coming from C# and java land I would have done it via functions, just thought this was cleaner or more how its supposed to be done in C++
I guess i can change it to a function.

Reply:
You just need to use an inline function :slightly_smiling_face: 
In this case, the compiler will surely optimize the call.
Macros can have unexpected side effects and can throws really weird errors, it's basically text replacement. It's more a C thing but in C++ we have been replacing them more and more: templates, constexpr, inline, variadic templates, ... :slightly_smiling_face:

Reply:
you can just define it like so:
```cpp
inline auto themeDebugLog() {
  return qDebug() << "[Theme]";
}
```